screen-write.c

```diff
@@ -653,34 +655,41 @@ screen_write_menu(struct screen_write_ctx *ctx, struct menu *menu,
 	cx = s->cx;
 	cy = s->cy;
 
-	memcpy(&default_gc, &grid_default_cell, sizeof default_gc);
+	memcpy(&default_gc, menu_gc, sizeof default_gc);
 
 	screen_write_box(ctx, menu->width + 4, menu->count + 2,
 	    BOX_LINES_DEFAULT, &default_gc, menu->title);
 
 	for (i = 0; i < menu->count; i++) {
 		name = menu->items[i].name;
 		if (name == NULL) {
+			/* Draw separator line */
 			screen_write_cursormove(ctx, cx, cy + 1 + i, 0);
-			screen_write_hline(ctx, menu->width + 4, 1, 1);
-		} else {
-			if (choice >= 0 && i == (u_int)choice && *name != '-')
-				gc = choice_gc;
-			screen_write_cursormove(ctx, cx + 2, cy + 1 + i, 0);
-			for (j = 0; j < menu->width; j++)
-				screen_write_putc(ctx, gc, ' ');
-			screen_write_cursormove(ctx, cx + 2, cy + 1 + i, 0);
-			if (*name == '-') {
-				name++;
-				default_gc.attr |= GRID_ATTR_DIM;
-				format_draw(ctx, gc, menu->width, name, NULL,
-				    0);
-				default_gc.attr &= ~GRID_ATTR_DIM;
-			} else
-				format_draw(ctx, gc, menu->width, name, NULL,
-				    gc == choice_gc);
-			gc = &default_gc;
+			screen_write_hline(ctx, menu->width + 4, 1, 1,
+			    &default_gc);
+			continue;
 		}
+
+		if (choice >= 0 && i == (u_int)choice && *name != '-')
+			gc = choice_gc;
+		screen_write_cursormove(ctx, cx + 2, cy + 1 + i, 0);
+		for (j = 0; j <= menu->width; j++)
+			screen_write_putc(ctx, gc, ' ');
+		screen_write_cursormove(ctx, cx + 1, cy + 1 + i, 0);
+		screen_write_putc(ctx, gc, ' ');
+		if (*name == '-') {
+			name++;
+			default_gc.attr |= GRID_ATTR_DIM;
+			format_draw(ctx, gc, menu->width, name, NULL, 0);
+			default_gc.attr &= ~GRID_ATTR_DIM;
+		} else
+			format_draw(ctx, gc, menu->width, name, NULL,
+			    // TODO: Should the following be 0 to always show
+			    // inline style even when the menu item is selected?
+			    gc == choice_gc);
```

Comment:
Does styling of choice take precedence of inline styling (e.g.: `display-menu '#[fg=cyan,bg=blue] using'`) ?
| Yes `gc == choice_gc` | No `0`
| --- | ---
| ![image](https://github.com/tmux/tmux/assets/16507/7c892a3a-c476-497a-b832-05d38e041654) | ![image](https://github.com/tmux/tmux/assets/16507/be96203f-488a-4f99-91ee-a3839dde7e9e)